frontend/src/model/beta/overview/FormEditPage.tsx

```diff
@@ -21,22 +22,24 @@ type FormEditPageProps = {
 export default function FormEditPage({ model }: FormEditPageProps) {
   const [isEdit, setIsEdit] = useState(false)
   const [splitSchema, setSplitSchema] = useState<SplitSchemaNoRender>({ reference: '', steps: [] })
-  const [dialogOpen, setDialogOpen] = useState(false)
 
   const { schema, isSchemaLoading, isSchemaError } = useGetSchema(model.card.schemaId)
   const { mutateModel } = useGetModel(model.id)
   const { mutateModelCardRevisions } = useGetModelCardRevisions(model.id)
   const { uiConfig: _uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
+  const [dialogOpen, setDialogOpen] = useState(false)
+  const [loading, setLoading] = useState(false)
 
   const { setUnsavedChanges } = useContext(UnsavedChangesContext)
 
   async function onSubmit() {
+    setLoading(true)
     if (schema) {
       const data = getStepsData(splitSchema, true)
       const res = await putModelCard(model.id, data)
       if (res.status && res.status < 400) {
         setIsEdit(false)
-
+        setLoading(false)
```

Comment:
This should be moved outside of this `if` statement. WE want to `setLoading(false)` when we get a response, regardless of what that response is.